service/k8sclient/common/core.go

```diff
@@ -119,12 +57,18 @@ const (
 )
 
 // IsContainerInState returns true if container is in give state, otherwise false.
-func IsContainerInState(containerStatuses []ContainerStatus, state ContainerState, containerName string) bool {
+func IsContainerInState(containerStatuses []corev1.ContainerStatus, state ContainerState, containerName string) bool {
+	containerState := make(map[string]interface{})
```

Comment:
is there some reason for not making the value here an empty struct? i.e:
```
containerState := make(map[string]struct{})
```

Reply:
We need it to marshal/unmarshal JSON